Build failure: wipe

## Steps To Reproduce

Steps to reproduce the behavior:

1. build `wipe` from unstable (`github:nixos/nixpkgs/634fd46801442d760e09493a794c4f15db2d0cbb`).

## Build log

<!-- insert build log in code block in collapsable section -->

<details>

<summary>Build Log</summary>

```
Running phase: unpackPhase
@nix { "action": "setPhase", "phase": "unpackPhase" }
unpacking source archive /nix/store/0fyqydvrgzq51lb8dbyfarx498pci3q4-wipe-2.3.1.tar.bz2
source root is wipe-2.3.1
setting SOURCE_DATE_EPOCH to timestamp 1257109891 of file wipe-2.3.1/wipe.h
Running phase: patchPhase
@nix { "action": "setPhase", "phase": "patchPhase" }
applying patch /nix/store/m4bw4dag08g4gfsxrxiyk45h9128bfmi-fix-install.patch
patching file Makefile.in
Running phase: updateAutotoolsGnuConfigScriptsPhase
@nix { "action": "setPhase", "phase": "updateAutotoolsGnuConfigScriptsPhase" }
Updating Autotools / GNU config script to a newer upstream version: ./config.sub
Updating Autotools / GNU config script to a newer upstream version: ./config.guess
Running phase: configurePhase
@nix { "action": "setPhase", "phase": "configurePhase" }
patching script interpreter paths in ./configure
./configure: interpreter directive changed from "#! /bin/sh" to "/nix/store/5mh7kaj2fyv8mk4sfq1brwxgc02884wi-bash-5.2p37/bin/sh"
configure flags: --prefix=/nix/store/kksykx8ipwcj2kjhyqxdh5pklf4g83rl-wipe-2.3.1
creating cache ./config.cache
checking host system type... x86_64-pc-linux-gnu
checking target system type... x86_64-pc-linux-gnu
checking build system type... x86_64-pc-linux-gnu
checking for gcc... gcc
checking whether the C compiler (gcc  ) works... no
configure: error: installation or configuration problem: C compiler cannot create executables.

```

</details>

## Additional context

I use `nixpkgs.hostPlatform.gcc.{arch,tune} = "znver4";`, which may be relevant.
Probably more so related to recent change for default gcc to 14.

## Metadata

<!-- Please insert the output of running `nix-shell -p nix-info --run "nix-info -m"` below this line -->

 - system: `"x86_64-linux"`
 - host os: `Linux 6.12.5-xanmod1, NixOS, 25.05 (Warbler), 25.05.20241219.d70bd19`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Lix, like Nix) 2.91.1
System type: x86_64-linux
Additional system types: i686-linux, x86_64-v1-linux, x86_64-v2-linux, x86_64-v3-linux, x86_64-v4-linux
Features: gc, signed-caches
System configuration file: /etc/nix/nix.conf
User configuration files: /home/alfarel/.config/nix/nix.conf:/nix/store/v5pkh7wda19iw13vsc4ma7cr8sbhzxs1-stylix-kde-config/nix/nix.conf:/etc/xdg/nix/nix.conf:/home/alfarel/.nix-profile/etc/xdg/nix/nix.conf:/nix/profile/etc/xdg/nix/nix.conf:/home/alfarel/.local/state/nix/profile/etc/xdg/nix/nix.conf:/etc/profiles/per-user/alfarel/etc/xdg/nix/nix.conf:/nix/var/nix/profiles/default/etc/xdg/nix/nix.conf:/run/current-system/sw/etc/xdg/nix/nix.conf
Store directory: /nix/store
State directory: /nix/var/nix
Data directory: /nix/store/3vdv3p0nhfzz4k3xv5dfmn6qw653xc2z-lix-2.91.1/share`
 - nixpkgs: `/nix/store/8fwsiv0hd7nw1brkvka0jf1frk3m7qkr-source`

## Notify maintainers

<!--
Please @ people who are in the `meta.maintainers` list of the offending package or module.
If in doubt, check `git blame` for whoever last touched something.
-->

@abbradar , but they haven't committed in 2 years, so not sure if they are still at all active

---

Note for maintainers: Please tag this issue in your PR.

---

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/by-name/wi/wipe/package.nix
+++ b/pkgs/by-name/wi/wipe/package.nix
@@ -16,6 +16,10 @@ stdenv.mkDerivation rec {
 
   nativeBuildInputs = [ autoreconfHook ];
 
+  # fdatasync is undocumented on darwin with no header file which breaks the build.
+  # use fsync instead.
+  configureFlags = lib.optional stdenv.hostPlatform.isDarwin "ac_cv_func_fdatasync=no";
+
   patches = [ ./fix-install.patch ];
 
   meta = with lib; {

--- a/pkgs/by-name/wi/wipe/package.nix
+++ b/pkgs/by-name/wi/wipe/package.nix
@@ -2,6 +2,7 @@
   lib,
   stdenv,
   fetchurl,
+  autoreconfHook,
 }:
 
 stdenv.mkDerivation rec {
@@ -12,6 +13,8 @@ stdenv.mkDerivation rec {
     url = "mirror://sourceforge/wipe/${version}/${pname}-${version}.tar.bz2";
     sha256 = "180snqvh6k6il6prb19fncflf2jcvkihlb4w84sbndcv1wvicfa6";
   };
+
+  nativeBuildInputs = [ autoreconfHook ];
 
   patches = [ ./fix-install.patch ];
 

